### PR TITLE
[MM-18118] Check locale is valid before sending to intl_provider

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
If an invalid locale is specified in the database, upon refreshing the app a logged in user will experience a blank screen. This PR forces the locale to the default in this case, such that `intl_provider` is never fed an invalid locale.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-18118

### CODE REVIEW

File: selectors/i18n.js [hunk 1]
```diff
@@ -3,12 +3,17 @@
...
 import {getCurrentUserLocale} from 'mattermost-redux/selectors/entities/i18n';
+import {General} from 'mattermost-redux/constants';
 
 import * as I18n from 'i18n/i18n';
 
 // This is a placeholder for if we ever implement browser-locale detection
 export function getCurrentLocale(state) {
-    return getCurrentUserLocale(state, getConfig(state).DefaultClientLocale);
+    const currentLocale = getCurrentUserLocale(state, getConfig(state).DefaultClientLocale);
+    if (I18n.isLanguageAvailable(currentLocale)) {
```
Comment (Willyfrog) [inline selectors/i18n.js]: shouldn't we check that there is a value in currentLocale?
```suggestion
if (currentLocale && I18n.isLanguageAvailable(currentLocale)) {
```

Comment (devinbinnie) [reply selectors/i18n.js]: `getCurrentUserLocale` already does a null check, it will return `DEFAULT_LOCALE` if the user's locale is null or empty.

**Comment by lindalumitchell** [comment]
Is this something that unit / e2e tests can and should be added for? @devinbinnie

**Comment by devinbinnie** [comment]
@lindalumitchell Definitely, I'll add a unit test to ensure an invalid user locale returns the default locale. Thanks for reminding me :)

**Pull request merged**